site/cypress/e2e/Cypress-Autograding/electronic_gradeable.spec.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Uploads a file and compares the results with expected scores.
+ * @param {string} fileUploadName the fixture to upload
+ * @param {number[]} fullScores the max scores for the gradeables
+ * @param {number[]} expectedScores the expected score for the submission
+ */
+const submitAndCheckResults = (fileUploadName, expectedScores, fullScores) => {
```

Comment:
let's go ahead and split this into two functions, submit & check results.
submit should verify that we are about to submit and create a specific version number, so we can do submit(allcorrect.zip,1) and submit(extra_lines_at_end.zip,1)
It only needs to wait to see that the upload was successful and that the version is available - it does not need to wait for autograding to complete.
and then check results will take in what versions grades should be checked: check_version_results(1, thefullcreditarray) and check_version_results(2,thepartialcreditvalues) and it will wait if the autograding isn't available.